src/System.Runtime/tests/System/Reflection/TypeInfoTests.cs

```diff
@@ -0,0 +1,263 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    public abstract class TestTypeBase : IDisposable
+    {
+        public abstract bool CanRead { get; }
+        public abstract bool CanWrite { get; }
+        public abstract bool CanSeek { get; }
+        public abstract long Length { get; }
+        public abstract long Position { get; set; }
+
+        public virtual Task FlushAsync()
+        {
+            throw null;
+        }
+
+        public abstract void Flush();
+
+        public virtual Task<int> ReadAsync(byte[] buffer, int offset, int count)
+        {
+            throw null;
+        }
+
+        public abstract int Read(byte[] buffer, int offset, int count);
+
+        public abstract long Seek(long offset, SeekOrigin origin);
+
+        public abstract void SetLength(long value);
+
+        public abstract void Write(byte[] buffer, int offset, int count);
+        public virtual Task WriteAsync(byte[] buffer, int offset, int count)
+        {
+            throw null;
+        }
+
+        public void Dispose()
+        {
+            throw null;
+        }
+    }
+
+    public class TestType : TestTypeBase
+    {
+        public TestType()
+        {
+        }
+
+        public class Nested
+        {
+
+        }
+
+#pragma warning disable 0067 // event never used
+        public event Action<int> StuffHappened;
+#pragma warning restore 0067
+
+#pragma warning disable 0169 // field never used
+        private int _pizzaSize;
+#pragma warning restore 0169
+
+        public override bool CanRead => false;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => false;
+
+        public override long Length
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override long Position
+        {
+            get
+            {
+                throw new NotImplementedException();
+            }
+
+            set
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        public override void Flush()
+        {
+            throw new NotImplementedException();
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void SetLength(long value)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    public class TypeInfoTests
+    {
+        public TypeInfo TypeInfo => typeof(TestType).GetTypeInfo();
```

Comment:
Naming this property "TypeInfo" is confusing. Might be preferable just to inline so that people can more easily read the tests in isolation.